Modules/core/BRPeer.c

```diff
@@ -275,6 +280,95 @@ static int _PeerAcceptVerackMessage(BRPeer *peer, const uint8_t *msg, size_t msg
     return r;
 }
 
+static int _PeerAcceptAssetMessage(BRPeer *peer, const uint8_t *msg, size_t msgLen) {
+    size_t off = 0, count = (size_t) BRVarInt(msg, msgLen, &off), sLen = 0;
+    int r = 1;
+    
+    if (off == 0) {
+        peer_log(peer, "malformed assets message");
+        r = 0;
+    } else if (msgLen > 16898) {
+        peer_log(peer, "dropping assets message, %zu is too many assets, max is 512", count);
+    } else {
+        peer_log(peer, "got asset with %zu data", count);
+        //        nameLen = UInt64GetLE(&msg[off]);
+        //        off += sizeof(uint64_t);
+        
+        BRAsset *asset = NewAsset();
+
+        asset->nameLen = count;
+        
+        asset->name = malloc(count + 1);
+        memcpy(asset->name, msg + off, count);
+        off += count;
+        *(asset->name + count) = '\0';
+        assert(*(asset->name + count) == '\0');
+        
+        asset->amount = (off + sizeof(uint64_t) <= msgLen) ? UInt64GetLE(&msg[off]) : 0;
+        off += sizeof(uint64_t);
+        
+        asset->unit = BRVarInt(&msg[off], (off <= (msgLen) ? (msgLen) - off : 0), &sLen);
+        off += sLen;
+        
+        asset->reissuable = BRVarInt(&msg[off], (off <= (msgLen) ? (msgLen) - off : 0), &sLen);
+        off += sLen;
+        
+        asset->hasIPFS = BRVarInt(&msg[off], (off <= (msgLen) ? (msgLen) - off : 0), &sLen);
+        off += sLen;
+        
+        size_t IPFS_length = (size_t) BRVarInt(&msg[off], (off <= (msgLen) ? (msgLen) - off : 0),
+                                               &sLen);
+        off += sLen;
+        
+        // Check the end of the script
+        if (asset->hasIPFS != 0 || IPFS_length != 0) {
+        
+        uint8_t IPFS_hash[IPFS_length];
+        
+        if (off <= msgLen + IPFS_length) {
+            memcpy(&IPFS_hash, msg + off, IPFS_length);
+            off += IPFS_length;
+            printf("\nIPFS hash: %s", IPFS_hash);
+            
+            EncodeIPFS(asset->IPFSHash, 47, IPFS_hash, IPFS_length);
+        }
+        }
+        
+        ((BRPeerContext *) peer)->receiveAssetData(peer->assetCallbackInfo, asset);
+    }
+    
+    return r;
+}
+
+static int _PeerAssetNotFoundMessage(BRPeer *peer, const uint8_t *msg, size_t msgLen) {
+    
+    size_t off = 0, size = (size_t) BRVarInt(msg, msgLen, &off), nameLen;
+    int r = 1;
+    
+    peer_log(peer, "got asset msg with %zu data", size);
+    
+    if (size == 0) {
+        peer_log(peer, "malformed assets message");
+        r = 0;
+    } else {
+        //TODO : assign name and len to new asset
+
+        nameLen = UInt64GetLE(&msg[off]);
+        off += sizeof(uint64_t);
+        
+        char name[nameLen];
```

Comment:
Should this be nameLen +1

Reply:
I will have to remove it anyway, on iOS I play on asset!=nil